.github/workflows/pytest.yaml

```diff
@@ -0,0 +1,36 @@
+---
+name: pytest
+
+on:
+  pull_request:
+    branches:
+      - main
+jobs:
+  pytest:
+    runs-on: ubuntu-latest
+
+    timeout-minutes: 15
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install .
+      - name: Test with pytest
+        run: |
+          pytest test \
+            --doctest-modules \
```

Comment:
Doctest looks interesting. Do we make use of it anywhere?

Reply:
I don't think so... full disclosure, that's straight from GitHub's example, I didn't actually consider it.